### PR TITLE
Add forward secrecy to State Proofs (WIP: SP key deletion)

### PR DESCRIPTION
**Pull request by algonathan**
## Context
In order to achieve forward security Stateproof's ephemeral keys must be deleted from the Database/registry after the round they are used in.

## Acceptance criteria
The node removes the ephemeral private keys after signing a state proof.

### CODE REVIEW

**Review by algoidan — COMMENTED** [review]
looks good. but we do need to change the events in participation registry.

File: data/account/registeryDbOps.go [hunk 1]
```diff
@@ -53,6 +54,35 @@ type appendKeysOp struct {
...
+
+		// Fetch primary key
+		var pk int
+		row := tx.QueryRow(selectPK, d.ParticipationID[:])
+		err := row.Scan(&pk)
+		if err != nil {
+			return fmt.Errorf("unable to scan pk: %w", err)
+		}
+
+		stmt, err := tx.Prepare(deleteStateProofKeysQuery)
```
Comment (algoidan) [inline data/account/registeryDbOps.go:73]: should we call close on prepare statement?

File: crypto/compactcert/builder_test.go [hunk 2]
```diff
@@ -121,10 +121,9 @@ func TestBuildVerify(t *testing.T) {
 	}
 
 	for i := 0; i < npart; i++ {
-		err = b.Add(uint64(i), sigs[i], !doLargeTest)
-		if err != nil {
-			t.Error(err)
-		}
+		a.False(b.Present(uint64(i)))
+		a.NoError(b.IsValid(uint64(i), sigs[i], !doLargeTest))
+		b.Add(uint64(i), sigs[i])
```
Comment (algoidan) [inline crypto/compactcert/builder_test.go:126]: let's make sure that Add didn't return ad error + there are several instances for that check

File: data/account/registeryDbOps.go [hunk 3]
```diff
@@ -53,6 +54,37 @@ type appendKeysOp struct {
...
+		defer stmt.Close()
```
Comment (algoidan) [inline data/account/registeryDbOps.go:74]: maybe it is better to close it after checking for errors.

**Review by algoidan — COMMENTED** [review]
looks great!
just a few nits

File: go.mod [hunk 4]
```diff
@@ -4,6 +4,7 @@ go 1.14
 
 require (
 	github.com/algorand/falcon v0.0.0-20220130164023-c9e1d466f123
+	github.com/algorand/go-algorand/cmd/partitiontest_linter v0.0.0-20220303022401-09b6c38e1262 // indirect
```
[the remark this answers is not in the document]
Comment (algonathan) [reply go.mod:7]: my bad

**Review by algoidan — APPROVED** [review]

**Pull request merged**